### ISSUE TITLE
XML.getChildren() returns additional, empty child XMLs

### ISSUE THREAD

**Issue by liquid600pgm** [processing/processing#5489]
## Description

Title.

## Expected Behavior

The expected output wouldn't have any empty XMLs.

## Current Behavior

If you look at the console log of the sketch below, it shows this:

```
[0] 
[1] <table><row><col>1,1</col><col>1,2</col></row><row><col>2,1</col><col>2,2</col></row><row><col>3,1</col><col>3,2</col></row></table>
[2] 
-----
:
<table><row><col>1,1</col><col>1,2</col></row><row><col>2,1</col><col>2,2</col></row><row><col>3,1</col><col>3,2</col></row></table>:
  row 0 '':
  row 1 '<row><col>1,1</col><col>1,2</col></row>':
    col 0 ''
    col 1 '<col>1,1</col>'
    col 2 ''
    col 3 '<col>1,2</col>'
    col 4 ''
  row 2 '':
  row 3 '<row><col>2,1</col><col>2,2</col></row>':
    col 0 ''
    col 1 '<col>2,1</col>'
    col 2 ''
    col 3 '<col>2,2</col>'
    col 4 ''
  row 4 '':
  row 5 '<row><col>3,1</col><col>3,2</col></row>':
    col 0 ''
    col 1 '<col>3,1</col>'
    col 2 ''
    col 3 '<col>3,2</col>'
    col 4 ''
  row 6 '':
:
```

## Steps to Reproduce

Run the following code:

```Processing
XML xml = loadXML("xml.xml");

printArray(xml.getChildren());
println("-----");

XML[] children = xml.getChildren();

for (int i = 0; i < children.length; i++) {
  println(children[i] + ":");
  XML[] rows = children[i].getChildren();
  for (int r = 0; r < rows.length; r++) {
    println("  row " + r + " '" + rows[r] + "'" +":");
    XML[] cols = rows[r].getChildren();
    for (int c = 0; c < cols.length; c++) {
      println("    col " + c + " '" + cols[c] + "'");
    }
  }
}

exit();
```

## Your Environment

 

* Processing version: 3.3.7
* Operating System and OS version: Windows 10 Fall Creators
* Other information: JRE 8u171; JDK 8u162

## Possible Causes / Solutions

Simply put, remove any empty XMLs in `getChildren()`.

**Comment by benfry** [comment]
Those are text elements, and are required in order to reconstruct the original XML with its original whitespace. You can use `trim()` to remove them.

**Comment by liquid600pgm** [comment]
@benfry `trim()` should be documented in the Javadoc, the doc comment is commented out, probably because of a mistake when re-writing code,

**Comment by benfry** [comment]
It'd be great if you could help out by submitting a PR.

**Comment by liquid600pgm** [comment]
Done. #5492

**Comment by benfry** [comment]
Thanks!
